data/image_train_item.py

```diff
@@ -111,8 +111,15 @@ def __get_weighted_shuffled_tags(seed: int, tags: list[str], weights: list[float
         return caption
 
     @staticmethod
-    def __get_shuffled_tags(seed: int, tags: list[str]) -> str:
-        random.Random(seed).shuffle(tags)
+    def __get_shuffled_tags(seed: int, tags: list[str], keep_tags: int) -> str:
+        tags = tags.copy()
+
+        if len(tags) > keep_tags:
```

Comment:
If the number of tags on some images is less than keep_tags, wouldn't this still shuffle them? 

Would it be better to do something like `local_keep_tags = max(len(tags)-1, keep_tags)` instead of the if statement?

Reply:
Or perhaps it just skips over and nothing gets shuffled right in the case len(tags) <= keep_tags ?

Reply:
The idea of `keep_tags` is basically freeze n number of tags at the begining. So if the image have less tags then nothing is changed.

What I think it should be covered is somebody puts a negative number there... probably `min(keep_tags, 0)`